index.js

```diff
@@ -32,7 +44,12 @@ module.exports = {
     if (options.liveReload !== true) { return; }
 
     process.env.EMBER_CLI_INJECT_LIVE_RELOAD_PORT = options.liveReloadPort;
-    process.env.EMBER_CLI_INJECT_LIVE_RELOAD_BASEURL = options.baseURL; // default is '/'
+    process.env.EMBER_CLI_INJECT_LIVE_RELOAD_BASEURL = options.baseURL;
+
+    if (options.detectLiveReloadOrigin === true) {
+      process.env.EMBER_CLI_INJECT_LIVE_RELOAD_HOST = options.host;
+      process.env.EMBER_CLI_INJECT_LIVE_RELOAD_ORIGIN = util.format('http://%s:%d/', options.host, options.port);
```

Comment:
I don't think we should hard-code the protocol here (it is possible to run `ember s` with SSL).